### PR TITLE
Add BitField objects for User & Application flags

### PR DESCRIPTION
**Pull request by panley01**
Also adds Application flag checks to the client code, checking priv. intents against Application flags to raise exceptions on unpermitted intents before initiating a connection with Discord.

### CODE REVIEW

**Review by i0bs — APPROVED** [review]
You have pleased me.

**Comment by panley01** [comment]
Amended code to account for `_intent` as opposed to `intent` & priv intent flags for unverified apps (_LIMITED suffix). Also moved intent checks before command checks for that sweet nanosecond optimizaton on knowing you fucked up

File: interactions/client.py [hunk 1]
```diff
@@ -256,6 +257,12 @@ async def _ready(self) -> None:
         ready: bool = False
 
         try:
+            if self._intents.GUILD_PRESENCES and not (self.me.flags.GATEWAY_PRESENCE or self.me.flags.GATEWAY_PRESENC_LIMITED):
```
Comment (Damego) [inline interactions/client.py]:
```suggestion
            if self._intents.GUILD_PRESENCES and not (self.me.flags.GATEWAY_PRESENCE or self.me.flags.GATEWAY_PRESENCE_LIMITED):
```

**Comment by i0bs** [comment]
Covering over Delta's review part, but the changes look proper and I haven't hit any issues in testing. LGTM. Thank you for contributing, gamer. 🤓

**Pull request merged**